app/gui/src/project-view/components/GraphEditor/GraphNode.vue

```diff
@@ -434,6 +438,20 @@ const nodeClass = computed(() => {
   }
 })
 
+const nodeProgress = computed(() => evaluationProgress(expressionInfo.value) ?? 100)
+const { getProgressStyles, watchProgress } = useProgressBackground(nodeProgress, {
+  counterId: () => expressionInfo.value?.evaluationId,
+})
+const { progressAnimating, backgroundProgressEvents } = watchProgress()
+
+// The progress bar animation and the color change animation are incompatible. When progress is not
+// partial or animating, we stop applying progress styles and fall back to the background color set
+// by the class, so that its CSS transition will be used if the type changes. (If the type changes
+// during progress animation, we'll just change colors abruptly).
```

Comment:
I think you could avoid the animation conflict and simplify the logic significantly by leveraging CSS `@property` declarations to declare our color custom properties as animated. That should allow them to be used inside `linear-gradient` definition, and in turn make it animated as well. The only difference between "in progress" and "static" state would be that one uses a gradient, and the other uses the color property directly. The selection color transition would then be happening on the property itself, independently of the progress state.